project-base/storefront/components/Pages/Customer/MergeCartsPopup.tsx

```diff
@@ -0,0 +1,32 @@
+import { Button } from 'components/Forms/Button/Button';
+import { Popup } from 'components/Layout/Popup/Popup';
+import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
+import { Dispatch, SetStateAction } from 'react';
+
+type MergeCartsPopupProps = {
+    mergeOrderItemsWithCurrentCart: (orderUuid: string, shouldMerge?: boolean | undefined) => void;
+    orderForPrefillingUuid: string;
+    setOrderForPrefillingUuid: Dispatch<SetStateAction<string | undefined>>;
+};
+
+export const MergeCartsPopup: FC<MergeCartsPopupProps> = ({
+    mergeOrderItemsWithCurrentCart,
+    orderForPrefillingUuid,
+    setOrderForPrefillingUuid,
+}) => {
+    const t = useTypedTranslationFunction();
+
+    return (
+        <Popup onCloseCallback={() => setOrderForPrefillingUuid(undefined)}>
```

Comment:
No need to pass whole setter function. Enough in this case would be pass simple void callback `() => void` and have setter function in the parent. 

Then we can call this callback function just `onClose` or similar.

Reply:
I hope I understood your request correctly... If not, let me know and I will fix it :slightly_smiling_face: